deepchem/models/tests/test_cgcnn.py

```diff
@@ -0,0 +1,58 @@
+import unittest
+from os import path, remove
+
+from deepchem.feat import CGCNNFeaturizer
+from deepchem.molnet import load_perovskite
+from deepchem.metrics import Metric, mae_score
+from deepchem.models import CGCNNModel, losses
+
+try:
+  import dgl  # noqa
+  import torch  # noqa
+  has_pytorch_and_dgl = True
+except:
+  has_pytorch_and_dgl = False
+
+
+@unittest.skipIf(not has_pytorch_and_dgl, 'PyTorch and DGL are not installed')
+def test_cgcnn():
+  # load datasets
+  current_dir = path.dirname(path.abspath(__file__))
+  config = {
+      "reload": False,
+      "featurizer": CGCNNFeaturizer,
+      # disable transformer
+      "transformers": [],
+      # load 'deepchem/models/test/perovskite.tar.gz'
+      "data_dir": current_dir
+  }
+  tasks, datasets, transformers = load_perovskite(**config)
+  train, valid, test = datasets
+
+  # initialize models
+  model = CGCNNModel(
+      in_node_dim=92,
+      hidden_node_dim=64,
+      in_edge_dim=41,
+      num_conv=3,
+      predicator_hidden_feats=128,
+      n_out=1,
+      loss=losses.L2Loss(),
+      batch_size=32,
+      learning_rate=0.001)
+
+  # train
+  model.fit(train, nb_epoch=50)
+  model.restore()
```

Comment:
Do we need to call `restore()` and `save_checkpoint()` for correct behavior here? It might be useful to factor out a separate test that `restore()`/`save_checkpoint()` work correctly

Reply:
`fit()` automatically saves a checkpoint at the end.  So calling `restore()` just reloads exactly the same parameters that are already in memory.  And then calling `save_checkpoint()` replaces the existing checkpoint with another identical one.  I think both those lines should be removed.

Reply:
Thank you for comments. I removed those lines.